sky/utils/common_utils.py

```diff
@@ -88,11 +94,46 @@ def _is_valid_user_hash(user_hash: Optional[str]) -> bool:
     return user_hash
 
 
-def truncate_and_hash_cluster_name(cluster_name: str) -> str:
-    if len(cluster_name) < 15:
-        return cluster_name
-    return cluster_name[:10] + hashlib.md5(
-        cluster_name.encode()).hexdigest()[:CLUSTER_NAME_HASH_LENGTH]
+def base36_encode(hex_str: str) -> str:
+    """Converts a hex string to a base36 string."""
+    int_value = int(hex_str, 16)
+
+    def _base36_encode(num: int) -> str:
+        if num == 0:
+            return '0'
+        alphabet = '0123456789abcdefghijklmnopqrstuvwxyz'
+        base36 = ''
+        while num != 0:
+            num, i = divmod(num, 36)
+            base36 = alphabet[i] + base36
+        return base36
+
+    return _base36_encode(int_value)
+
+
+def truncate_and_hash_cluster_name(cluster_name: str,
+                                   max_length: Optional[int] = 15,
+                                   add_user_hash: bool = True) -> str:
+    user_hash = ''
+    if add_user_hash:
+        user_hash = get_user_hash()[:USER_HASH_LENGTH_IN_CLUSTER_NAME]
+        user_hash = f'-{user_hash}'
+    user_hash_length = len(user_hash)
+
+    if (max_length is None or
```

Comment:
add a doc string for meaning when `max_length is None`?

Reply:
Good point! Added the docstr. Thanks!